zerver/tests/test_reactions.py

```diff
@@ -265,6 +256,21 @@ def test_remove_nonexisting_reaction(self):
                                     **self.api_auth(reaction_sender))
         self.assert_json_error(second, "Reaction does not exist")
 
+    def test_remove_existing_reaction_with_renamed_emoji(self):
+        # type: () -> None
+        """
+        Removes an old existing reaction but the name of emoji got changed during
+        various emoji infra changes.
+        """
+        sender = self.example_email("hamlet")
+        result = self.client_put('/api/v1/messages/1/emoji_reactions/smile',
+                                 **self.api_auth(sender))
+        self.assert_json_success(result)
+
+        with mock.patch('zerver.lib.emoji.name_to_codepoint', name_to_codepoint={}):
+            result = self.client_delete('/api/v1/messages/1/emoji_reactions/smile',
+                                     **self.api_auth(sender))
+            self.assert_json_success(result)
```

Comment:
We should probably add a version of this test for a deactivated realm emoji, since that's the other common case we're trying to handle here.